### ISSUE TITLE
Problems with using-style typedefs containing template types

### ISSUE THREAD

**Issue by egraether** [CoatiSoftware/CoatiBugTracker#128]
```
template<class Vertex>
class Mesh
    : public Model<Vertex>
{
public:
    using Vertices = typename Model<Vertex>::Vertices;

    Mesh(Vertices vertices, Indices indices) {...}
}
```

**Comment by upsj** [comment]
Minimal example:
`template<class T>
class A {
public:
        using Type = int;
};

template<class T>
class B : public A<T> {
public:
        using Type = typename A<T>::Type;
};

int main() {
        B<int> test;
}`

The type of A<int>::Type (the second one at the bottom -- why is there a second "Type" entry in the diagram?) and A<B<class T>::T>::Type cannot be deduced, even though they are independent of T.

My Coati version is 0.7.0.22

But I understand that this kind of type deduction is quite a difficult task to solve in general, so only consider it if there is a reasonable solution ;)

**Comment by mlangkabel** [comment]
@upsj thank you very much for providing that example! I'll take a look at it as soon as possible.

**Comment by mlangkabel** [comment]
Fixed in Coati 0.7.1.12
